app/src/main/java/com/expidev/gcmapp/MeasurementDetailsActivity.java

```diff
@@ -82,94 +90,59 @@ protected void onCreate(Bundle savedInstanceState)
         super.onCreate(savedInstanceState);
         setContentView(R.layout.base_graph);
 
-        preferences = getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
-
-        String measurementId = getIntent().getStringExtra("measurementId");
-        String ministryId = getIntent().getStringExtra("ministryId");
-        String mcc = getIntent().getStringExtra("mcc");
-        String period = getIntent().getStringExtra("period");
+        measurementId = getIntent().getStringExtra("measurementId");
+        ministryId = getIntent().getStringExtra("ministryId");
+        mcc = getIntent().getStringExtra("mcc");
+        period = getIntent().getStringExtra("period");
         measurementName = getIntent().getStringExtra("measurementName");
         ministryName = getIntent().getStringExtra("ministryName");
-
-        MeasurementsService.retrieveDetailsForMeasurement(this, measurementId, ministryId, mcc, period);
     }
 
     @Override
     protected void onStart()
     {
         super.onStart();
-        setupBroadcastReceivers();
+        setupLoaders();
     }
 
-    private void setupBroadcastReceivers()
+    private void setupLoaders()
     {
-        broadcastManager = LocalBroadcastManager.getInstance(this);
-
-        broadcastReceiver = new BroadcastReceiver()
-        {
-            @Override
-            public void onReceive(Context context, Intent intent)
-            {
-                if (BroadcastUtils.ACTION_START.equals(intent.getAction()))
-                {
-                    Log.i(TAG, "Action Started");
-                }
-                else if (BroadcastUtils.ACTION_RUNNING.equals(intent.getAction()))
-                {
-                    Log.i(TAG, "Action Running");
-                }
-                else if (BroadcastUtils.ACTION_STOP.equals(intent.getAction()))
-                {
-                    Log.i(TAG, "Action Done");
+        final LoaderManager manager = this.getSupportLoaderManager();
 
-                    Type type = (Type) intent.getSerializableExtra(BroadcastUtils.ACTION_TYPE);
+        Bundle args = new Bundle(4);
+        args.putString("measurementId", measurementId);
+        args.putString("ministryId", ministryId);
+        args.putString("mcc", mcc);
+        args.putString("period", period);
```

Comment:
these strings are being referenced in multiple places (and classes), it would be beneficial to use shared static constants to guarantee they are consistent everywhere they are being used.